warehouse/redshift.go

```diff
@@ -295,3 +325,56 @@ func (rs *Redshift) DoesTableExist(name string) bool {
 	}
 	return (exists != 0)
 }
+
+func (rs *Redshift) getTableColumns(name string) []string {
+	ctx := context.Background()
+	log.Printf("Fetching columns for table %s", name)
+	rows, err := rs.conn.QueryContext(ctx,
+		"SELECT \"column\" FROM pg_table_def WHERE schemaname = 'public' AND tablename = $1;", name)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var columns []string
+
+	defer rows.Close()
+	for rows.Next(){
+		var column string
+		err = rows.Scan(&column)
+		columns = append(columns, column)
+	}
+
+	// get any error encountered during iteration
+	if err = rows.Err(); err != nil {
+		log.Fatal(err)
+	}
+	return columns
+}
+
+func (rs *Redshift) getAlterTableStatement(f WarehouseField, tableName string) string {
+	return fmt.Sprintf("ALTER TABLE %s ADD COLUMN %s %s;", tableName, f.Name, f.DBType)
+}
+
+func (rs *Redshift) getMissingFields(schema Schema, tableColumns []string) []WarehouseField {
+	existingColumns := make(map[string]bool)
+	for _, column := range tableColumns {
+		// Redshift columns are case insensitive
+		existingColumns[strings.ToLower(column)] = true
+	}
+
+	var missingFields []WarehouseField
+	for _, f := range schema {
+		if _, ok := existingColumns[strings.ToLower(f.Name)]; !ok {
+			missingFields = append(missingFields, f)
+		}
+	}
+
+	return missingFields
+}
+
+func(rs *Redshift) getSchemaHeaders(schema Schema) []string {
+	var headers []string
+	for _, f := range schema {
+		headers = append(headers, f.Name)
+	}
+	return headers
+}
```

Comment:
nit: Please do add the last newline to all source files; that's pretty standard for all code at FS. A lot of times we don't explicitly point them out in code reviews since GitHub does such a good job of it; we just assume you can see it yourself and fix it.

Reply:
Just set up my (new) IDE to add the last newline to all source files, on save.